clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppNotification.java

```diff
@@ -452,33 +458,54 @@ boolean isTablet() {
         return isTablet;
     }
 
-    void prepareForDisplay(InAppResourceProvider inAppResourceProvider) {
+    void prepareForDisplay(
+            FileResourceProvider fileResourceProvider,
+            final TemplatesManager templatesManager
+    ) {
+        if (inAppType.equals(CTInAppType.CTInAppTypeCustomCodeTemplate)) {
+            final List<Pair<String, CtCacheType>> fileUrlMetas = customTemplateData.getFileArgsUrls(templatesManager);
 
-        for (CTInAppNotificationMedia media : this.mediaList) {
-            if (media.isGIF()) {
-                byte[] bytes = inAppResourceProvider.fetchInAppGif(media.getMediaUrl());
-                if (bytes != null && bytes.length > 0) {
-                    listener.notificationReady(this);
-                    return;
-                } else {
-                    this.error = "Error processing GIF";
-                }
-            } else if (media.isImage()) {
-
-                Bitmap bitmap = inAppResourceProvider.fetchInAppImage(media.getMediaUrl());
-                if (bitmap != null) {
-                    listener.notificationReady(this);
-                    return;
-                } else {
-                    this.error = "Error processing image as bitmap was NULL";
+            int index = 0;
+            while (index < fileUrlMetas.size()) {
+                Pair<String, CtCacheType> data = fileUrlMetas.get(index);
+                byte[] bytes = fileResourceProvider.fetchFile(data.getFirst());
+
+                if (bytes == null) {
+                    // download fail
+                    this.error = "some error";
```

Comment:
We need to discuss how to handle failed files with product. We could leave failed files as "null" and still display the in-app with no error. 

Meanwhile could we change the error string to something more descriptive. "Failed to download file" or something.

Reply:
Error is fixed